elibrary-micro/src/main/java/com/javafee/elibrary/micro/infrastructure/exception/CustomGlobalExceptionHandler.java

```diff
@@ -0,0 +1,95 @@
+import org.springframework.web.context.request.WebRequest;
+import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
+
+import lombok.extern.slf4j.Slf4j;
+
+@RestControllerAdvice
+@Slf4j
+public class CustomGlobalExceptionHandler extends ResponseEntityExceptionHandler {
+	// error handler for @Valid
+	@Override
+	protected ResponseEntity<Object> handleMethodArgumentNotValid(MethodArgumentNotValidException e,
+	                                                              HttpHeaders headers,
+	                                                              HttpStatus status, WebRequest request) {
+		log.error(getEndpointData(request) + e.getLocalizedMessage());
+		Map<String, Object> body = prepareResponseMap(e, status);
+		return new ResponseEntity<>(body, headers, status);
+	}
+
+	// error handler for Exception, code INTERNAL_SERVER_ERROR
+	@ExceptionHandler(Exception.class)
+	@ResponseStatus(HttpStatus.INTERNAL_SERVER_ERROR)
+	public ResponseEntity<?> exception(Exception e, WebRequest request) {
+		log.error(getEndpointData(request) + e.getLocalizedMessage());
+		return new ResponseEntity<>(prepareCustomExceptionDto(e, HttpStatus.INTERNAL_SERVER_ERROR, Collections.singletonList(e.getMessage()),
+				request.getDescription(false), ((ServletWebRequest) request).getHttpMethod().toString()),
+				HttpStatus.INTERNAL_SERVER_ERROR);
+	}
+
+	@ExceptionHandler(ConstraintViolationException.class)
+	@ResponseStatus(HttpStatus.BAD_REQUEST)
+	public ResponseEntity<?> handleConstraintViolationException(ConstraintViolationException e, WebRequest request) {
+		log.error(getEndpointData(request) + e.getLocalizedMessage());
+		return new ResponseEntity<>(prepareCustomExceptionDto(e, HttpStatus.BAD_REQUEST, Collections.singletonList(e.getMessage()),
+				request.getDescription(false), ((ServletWebRequest) request).getHttpMethod().toString()),
+				HttpStatus.BAD_REQUEST);
+	}
+
+	private String getEndpointData(WebRequest request) {
+		return MessageFormat.format(SystemProperties.getResourceBundle().getString("aspects.logTemplate.customGlobalExceptionHandler"),
+				((ServletWebRequest) request).getHttpMethod().toString(), request.getDescription(false));
+	}
+
+	private Map prepareResponseMap(Exception e, HttpStatus status) {
+		Map<String, Object> body = new LinkedHashMap<>();
+		body.put("timestamp", new Date());
+		body.put("status", status.value());
+
+		//Get all errors
+		List<String> errors = new ArrayList<>();
+		if (e instanceof MethodArgumentNotValidException)
+			errors = ((MethodArgumentNotValidException) e).getBindingResult()
+					.getFieldErrors()
+					.stream()
+					.map(DefaultMessageSourceResolvable::getDefaultMessage)
+					.collect(Collectors.toList());
+		else
+			errors.add(e.getMessage());
+		body.put("errors", errors);
+		return body;
+	}
+
+	private JfsExceptionDto prepareCustomExceptionDto(Exception e, HttpStatus status, List<String> messages, String path, String method) {
```

Comment:
Please change the object name from `JfsExceptionDto` to `ElibraryMicroExceptionDto`.

Reply:
done